### PR TITLE
Tests for editing addons with invalid payloads - part two

### PR DESCRIPTION
**Pull request by AlexandraMoga**
This PR includes:
- a test for editing addon names with a trademark name (Mozilla/Firefox)
- test for invalid addon homepage values
- test for addon invalid email domains
- test for invalid `experimental` and `requires payment` fields
- test for valid/invalid contribution domains
- test for invalid addon tags 

Edit: failing test is unrelated; caused by a basket email service disruption

### CODE REVIEW

File: tests/api/test_addon_uploads.py [hunk 1]
```diff
@@ -962,6 +962,44 @@ def test_extension_invalid_name(base_url, session_auth):
...
 
 
+@pytest.mark.parametrize(
+    'trademark_name',
+    [
+        'A Name with Firefox',
+        'A Name with Mozilla',
+        'A Name with Mozilla Firefox',
+        'Name has FireFox',
+        'Name has MOZILLA',
```
Comment (jrbenny35) [inline tests/api/test_addon_uploads.py:972]: Could we just do "Mozilla Firefox"? Do we know the exact filter for this?

File: tests/api/test_addon_uploads.py [hunk 2]
```diff
@@ -1002,6 +1040,265 @@ def test_extension_invalid_summary(base_url, session_auth):
...
+    """Try to add some invalid and unaccepted homepage urls for an addon"""
+    addon = payloads.edit_addon_details['slug']
+    invalid_homepage = [
+        '',
+        '.',
+        'abc123',
+        'example.com',
+        'http://not-valid',
+        'www.some-url.org',
+        base_url,
```
Comment (jrbenny35) [inline tests/api/test_addon_uploads.py:1055]: Do we need to test all of these?

**Review by jrbenny35 — CHANGES_REQUESTED** [review]
I see a lot of urls, or multiple inputs for tests. Are these all needed? Like for the contribution URL, do we need to test all of the accepted URLs?

File: tests/api/test_addon_uploads.py (hunk 1 above)
Comment (AlexandraMoga) [reply tests/api/test_addon_uploads.py:972]: We want to cover as many edge cases as possible, this is why I'm using all these combinations. For example, we want to make sure that all or some uppercase letters do not impact the detection (we had issues in the past when upper case letters caused issues in some areas, so better safe than sorry). Also, we want to make sure that the trademark name is detected both in isolation and when combined with another trademark.

File: tests/api/test_addon_uploads.py (hunk 2 above)
Comment (AlexandraMoga) [reply tests/api/test_addon_uploads.py:1055]: As before, yes, we'd like to cover as many edge cases as possible. We need to make sure that the API responds accordingly when provided with invalid or unaccepted values.

**Comment by AlexandraMoga** [comment]
> I see a lot of urls, or multiple inputs for tests. Are these all needed? Like for the contribution URL, do we need to test all of the accepted URLs?

@jrbenny35 I've replied to your review comments separately. As for the contribution URLs, again, we need to test that all the options made available for developers are working. As for the invalid contributions, there's not much I could cut there either. 
If I was to do a manual regression on these endpoints, I'd have to test all these combinations, so it would make sense to have them covered by the automated tests instead.

**Review by jrbenny35 — APPROVED** [review]
Thanks for answering my questions.
